src/session.rs

```diff
@@ -1082,7 +1111,8 @@ impl Session {
     pub fn edit<P: AsRef<Path>>(&mut self, paths: &[P]) -> io::Result<()> {
         // TODO: Keep loading paths even if some fail?
         for path in paths {
-            let path = path.as_ref();
+            let path_osstring = &self.expand_path(path);
+            let path = Path::new(path_osstring);
```

Comment:
I think it would make more sense for path expansion to happen in one place in the command parser (`cmd.rs`). You could add a method to `Parser` called `path()` which parses a path and expands the `~`.

Reply:
Then all paths handled by the session would already be expanded.

Reply:
I have since moved all of the logic into Parser from `cmd.rs`.